radix-engine/src/kernel/actor.rs

```diff
@@ -28,6 +29,27 @@ pub enum Actor {
 }
 
 impl Actor {
+    pub fn is_auth_zone(&self) -> bool {
+        match self {
+            Actor::Method(MethodActor { object_info, .. }) => {
+                object_info.blueprint.package_address.eq(&RESOURCE_PACKAGE)
+                    && object_info.blueprint.blueprint_name.eq(AUTH_ZONE_BLUEPRINT)
+            }
+            Actor::Function { .. } => false,
+            Actor::VirtualLazyLoad { .. } => false,
+            Actor::Root { .. } => false,
+        }
+    }
+
+    pub fn is_barrier(&self) -> bool {
+        match self {
+            Actor::Method(MethodActor { object_info, .. }) => object_info.global,
```

Comment:
I assume this also includes calls to modules:

I might be tempted to make that more explicit, ie: `module_id != Main || object_info.global`

Reply:
`object_info.global = true` is set for any object module in a global node.

Reply:
Not that we have them, but I imagine it should also be true for non-main modules in an internal node?